File: CPE/CPE49/25-UVA10093.cpp
#include <bits/stdc++.h>
using namespace std;
#define endl '\n'

int main() {
    ios::sync_with_stdio(false); cin.tie(nullptr); cout.tie(nullptr);
    int ans, s, mx;
    string line;
    while (cin >> line) {
        vector<int> N;
        for (char ch : line) {
            if (isupper(ch)) N.push_back(ch - 'A' + 10);
            else if (islower(ch)) N.push_back(ch - 'a' + 36);
            else if (isdigit(ch)) N.push_back(ch - '0');
        }
        s = 0;
        mx = -1;
        for (int x : N){
            mx = max(mx, x);
            s += x;
        }
        ans = -1;
        for (int i = max(2, mx + 1) ; i <= 62; ++i) {
            if (s % (i - 1) == 0) {
                ans = i;
                break;
            }
        }
        cout << (ans == -1 ? "such number is impossible!" : to_string(ans)) << endl;
    }
    return 0;
}

File: CPE/CPE49/25-UVA10093.py
"""
    進制轉換
    base-10下若一個數字的數位和為 9 的倍數，則該數字也是9的倍數
    可推廣到 base-r 下，若一個數字的每位總和為 r-1 的倍數，則該數字也是 r-1 的倍數

    翻了比較古老的翻譯，測資可能是包含非合法字元的，所以要先判斷是不是英文或數字
"""

while True:
    try:
        # N = list(input().strip())
        line = input().strip()
    except EOFError:
        break
    # for i, x in enumerate(N):
    #     if ord("A") <= ord(x) <= ord("Z"):
    #         N[i] = ord(x) - ord("A") + 10
    #     elif ord("a") <= ord(x) <= ord("z"):
    #         N[i] = ord(x) - ord("a") + 36
    #     else:
    #         N[i] = int(x)
    N = []
    for ch in line:
        if ch.isupper():
            N.append(ord(ch) - ord("A") + 10)
        elif ch.islower():
            N.append(ord(ch) - ord("a") + 36)
        elif ch.isdigit():
            N.append(int(ch))

    ans = -1
    r = max(N) + 1 # 最小必須為 base-r
    s = sum(N) 
    for i in range(max(2, r), 63):
        if s % (i-1) == 0:
            ans = i
            break
    print(ans if ans != -1 else "such number is impossible!")